### PR TITLE
Create an api for querying the event log.

### PR DESCRIPTION
**Pull request by vdbergh**
A valid request consists of a POST request with a body that is a stringified version of the json query.

The response is limited to 200 events.

To test the api there is an example document

<server>/html/finished_runs.html

This will be removed later.

### CODE REVIEW

**Comment by ppigazzini** [comment]
Updated.
https://dfts-0.pigazzini.it/html/finished_runs.html

File: server/fishtest/static/html/finished_runs.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html>
```
Comment (ppigazzini) [inline server/fishtest/static/html/finished_runs.html:2]: <html lang="en"> :D

**Comment by vdbergh** [comment]
It seems the database on DEV is not recent enough. There aren't any finished runs... We need to create some.

**Comment by ppigazzini** [comment]
I'm loading the latest DB backup.

File: server/fishtest/static/html/finished_runs.html (hunk 1 above)
Comment (vdbergh) [reply server/fishtest/static/html/finished_runs.html]: Fixed.

**Comment by ppigazzini** [comment]
Done!

**Comment by vdbergh** [comment]
It works now... Perhaps we can leave the page html/finished_runs.html for a while as an example for people that want to play with the api? What I want to do is to generate notifications for finished runs.

**Comment by ppigazzini** [comment]
It's fine for me, perhaps better to add a comment in the html page.
Not easy after some time to clean up the old scripts, old dbs, old collections,

**Comment by vdbergh** [comment]
I made the test page auto refreshing.

**Comment by vdbergh** [comment]
Comment added in test file.

File: server/fishtest/static/html/finished_runs.html [hunk 2]
```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head> </head>
+  <body>
+    <script>
+      // Example script to demonstrate the api.
```
[the remark this answers is not in the document]
Comment (ppigazzini) [reply server/fishtest/static/html/finished_runs.html:6]: No, sorry, it's in the script tag

**Comment by ppigazzini** [comment]
DEV updated.

**Comment by ppigazzini** [comment]
The 1000 Elo is an upper bound, I suppose.

**Comment by ppigazzini** [comment]
Auto refresh works.

**Comment by vdbergh** [comment]
> The 1000 Elo is an upper bound, I suppose.

Yes. The SPRT formulas don't work for extremely big Elo differences.

**Review by ppigazzini — APPROVED** [review]

**Comment by ppigazzini** [comment]
PROD updated, posted a message on Discord, thank you @vdbergh :)

**Comment by vdbergh** [comment]
Bummer. In retrospect it was not a good idea to store binary data in the actiondb (e.g. the time field). A generic remote query on the value of the time field is now impossible since json does not support the date format. 

I guess another db conversion would be best sometime.

**Comment by ppigazzini** [comment]
We have a beautiful maintenance page with a timer :)

**Comment by dav1312** [comment]
> We have a beautiful maintenance page with a timer :)

We can now also see it here: https://tests.stockfishchess.org/html/maintenance.html

**Comment by ppigazzini** [comment]
I added the html location to the nginx config for the example page of this PR. Before it was not reachable.

**Pull request merged**